src/server/pfs/server/driver.go

```diff
@@ -945,14 +945,14 @@ func (d *driver) repoSize(ctx context.Context, txnCtx *txncontext.TransactionCon
 }
 
 // propagateBranches starts commits downstream of 'branches'
-// in order to restore the invariant that branch provenance matches HEAD commit
+// in order to restore the invariant that branch provenance implies HEAD commit
 // provenance:
 //
-//	B.Head is provenant on A.Head <=>
-//	branch B is provenant on branch A
+//	branch B is provenant on branch A =>
+//	(B.Head is provenant on A.Head) OR (A.Head is nil)
```

Comment:
Hmm, with the new database migration, a branch head should never be nil. When a freshly minted branch is created, we create an empty sentinel commit for its head to point to.